src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,162 @@
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public final boolean equals(Object o) {
+            if (o == this) {
+                return true;
+            }
+            if (o.getClass().equals(Node.class)) {
+                Node e = (Node) o;
+                return e.key.equals(key) && e.value.equals(value);
+            }
+            return false;
+        }
+    }
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
+        if (table == null) {
+            resize();
+        }
+        int hash = getHash(key);
+
+        Node<K, V> node = new Node<>(hash, key, value, null);
+
+        Node<K, V> nodeByHash = table[hash];
+        boolean increaseSize;
+        if (nodeByHash == null) {
+            table[hash] = node;
+            increaseSize = true;
+        } else {
+            increaseSize = putNextOrChange(nodeByHash, node);
+        }
 
+        if (increaseSize && ++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (table == null) {
+            return null;
+        }
+
+        Node<K, V> node = table[getHash(key)];
+
+        if (node == null) {
+            return null;
+        } else {
+            do {
+                if (node.key == key
+                        || (node.key != null
+                        && node.key.equals(key)
+                        && node.key.hashCode() == key.hashCode())) {
+                    return node.value;
+                }
+
+                node = node.next;
+            } while (node != null);
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putData(Node<K, V> data) {
+        int hash = getHash(data.key);
+        data.hash = hash;
+
+        Node<K, V> nodeByHash = table[hash];
+
+        if (nodeByHash == null) {
+            table[hash] = data;
+        } else {
+            putNextOrChange(nodeByHash, data);
+        }
+    }
+
+    private boolean putNextOrChange(Node<K, V> src, Node<K, V> element) {
+        Node<K, V> curr = src;
+        Node<K, V> next;
+        K elementKey = element.key;
+
+        do {
+            next = curr.next;
+
+            if (curr.key == elementKey
+                    || (curr.key != null
+                    && curr.key.equals(elementKey)
+                    && curr.key.hashCode() == elementKey.hashCode())) {
```

Comment:
according to equals and hashcode contract - if A equals B, they always have the same hashcode, the opposite isn't guaranteed to be true